app/api/helpers/payment.py

```diff
@@ -200,6 +200,46 @@ def create_payment(order, return_url, cancel_url):
         else:
             return False, payment.error
 
+    @staticmethod
+    def verify_payment(payment_id, order):
+        """
+        Verify Paypal payment one more time for paying with Paypal in mobile client
+        """
+        try:
+            payment_server = paypalrestsdk.Payment.find(payment_id)
+            if payment_server.state != 'approved':
+                return False, 'Payment has not been approved yet. Status is ' + payment_server.state + '.'
+
+            # Get the most recent transaction
+            transaction = payment_server.transactions[0]
+            amount_server = transaction.amount.total
+            currency_server = transaction.amount.currency
+            sale_state = transaction.related_resources[0].sale.state
+
+            if amount_server != order.amount:
+                return False, 'Payment amount does not match order'
+            elif currency_server != order.event.payment_currency:
+                return False, 'Payment currency does not match order'
+            if sale_state != 'completed':
+                return False, 'Sale not completed'
+            elif PayPalPaymentsManager.used_payment(payment_id):
+                return False, 'Payment already been verified'
+            else:
+                return True, None
+        except paypalrestsdk.ResourceNotFound:
+            return False, 'Payment Not Found'
+
+    @staticmethod
+    def used_payment(payment_id):
+        """
+        Make sure same payment does not reused
```

Comment:
`function to check for recycling of payment IDs`

Reply:
>Orders should never be created with a paypal_token. Consider this as imitating a paid order through PayPal. After an order is created, only after proper authorization from PayPal's API, the attribute should be updated.
All you need to do is verify it right? So please remove this.

So how should I save a used payment ID? You told me to save it in the Order model, please clarify this for me? Thank you